enterprise/api/v1/serializers.py

```diff
@@ -191,37 +244,35 @@ def save(self):  # pylint: disable=arguments-differ
         ecu.save()
 
 
-class EnterpriseCustomerUserEntitlementSerializer(serializers.Serializer):
+class EnterpriseCustomerUserEntitlementSerializer(ImmutableStateSerializer):
     """
     Serializer for the entitlements of EnterpriseCustomerUser.
 
     This Serializer is for read only endpoint of enterprise learner's entitlements
     It will ignore any state changing requests like POST, PUT and PATCH.
     """
+
     entitlements = serializers.ListField(
         child=serializers.DictField()
     )
 
-    def create(self, validated_data):
-        """
-        Do not perform any operations for state changing requests.
-        """
-        pass
-
-    def update(self, instance, validated_data):
-        """
-        Do not perform any operations for state changing requests.
-        """
-        pass
     user = UserSerializer(read_only=True)
     enterprise_customer = EnterpriseCustomerSerializer(read_only=True)
     data_sharing_consent = UserDataSharingConsentAuditSerializer(many=True, read_only=True)
 
 
-class EnterpriseCourseCatalogReadOnlySerializer(serializers.Serializer):
+class EnterpriseCustomerCatalogApiPaginatedResponseReadOnlySerializer(ResponsePaginationSerializer):
```

Comment:
I'm in favor of the shorter name proposed by @itsjeyd - mostly because it isn't as important to be explicit about the pagination, I'm ok with it being implied as it isn't an unreasonable implication.